tests/data/metadata-injector-pod.yaml

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Debug pod for injecting GPU metadata to node's host filesystem
+# This mimics what metadata-collector does but for testing purposes
+apiVersion: v1
```

Comment:
Rather than creating a pod that sleeps, exec and then write metadata, can we instead create a config map, mount it to the pod and then pod can simply do a cp? That will be faster and also cleaner I feel